### PR TITLE
Advice journey - all the code for advice tickets

### PR DESCRIPTION
**Pull request by jon-edery**
This PR contain all the code for completed advice tickets.  This is not ready to be merged but will be used as a branch to deploy for testing.

Contains the following tickets:

PAR-934
PAR-844
PAR-1379
PAR-1380
PAR-1374
PAR-1158
PAR-1352

Screen Shot 2019-04-30 at 22 57 36

Screen Shot 2019-04-30 at 22 58 43

Screen Shot 2019-04-30 at 22 58 55

Screen Shot 2019-04-30 at 22 59 35

Screen Shot 2019-04-30 at 23 00 09

### CODE REVIEW

File: web/modules/custom/par_data/par_data.install [hunk 1]
```diff
@@ -558,6 +558,55 @@ function par_data_update_8032() {
...
+ */
+function par_data_update_8034() {
+  \Drupal::service('entity.definition_update_manager')->applyUpdates();
+}
+
+/**
+ * Setup config updates in par_partnership_flows.
+ */
+function par_data_update_8035() {
+  \Drupal::service('config.installer')->installDefaultConfig('module', 'par_partnership_flows');
```
Comment (kalpaitch) [inline web/modules/custom/par_data/par_data.install]: Why is this hook here? If the module is already installed you'll need to do a feature revert all to get it into the config `/sync` folder and then commit from there. This should be removed.

File: sync/core.extension.yml [hunk 2]
```diff
@@ -107,7 +108,6 @@ module:
   queue_ui: 0
   raven: 0
   rdf: 0
-  redis: 0
```
Comment (kalpaitch) [inline sync/core.extension.yml]: @jon-edery WTF???

File: sync/views.view.helpdesk_dashboard.yml [hunk 3]
```diff
@@ -2273,22 +2271,6 @@ display:
         weight: 0
         context: '0'
         menu_name: account
-      header:
```
Comment (kalpaitch) [inline sync/views.view.helpdesk_dashboard.yml]: WTF, also why are you removing this, I think you didn't merge config conflicts properly. I'm going to revert view changes.

File: sync/views.view.partnership_search.yml [hunk 4]
```diff
@@ -10,12 +10,13 @@ dependencies:
     - datetime
```
Comment (kalpaitch) [inline sync/views.view.partnership_search.yml]: I don't get what the changes are here, it seems that you're amending the partnership search view, and I think this is where the error is occurring. Worth a look at this and a chat.

File: sync/views.view.helpdesk_dashboard.yml (hunk 3 above)
Comment (jon-edery) [reply sync/views.view.helpdesk_dashboard.yml]: yea this was a merge conflict I resolved this should not be removed is an error.

File: sync/views.view.partnership_search.yml (hunk 4 above)
Comment (jon-edery) [reply sync/views.view.partnership_search.yml]: The changes I made to the partner search view are the features in -> https://regulatorydelivery.atlassian.net/browse/PAR-1352

Comment (jon-edery) [reply sync/views.view.partnership_search.yml]: this was to extend the partnership search view to include advice 
Screen Shot 2019-06-05 at 09 16 14

**Comment by kalpaitch** [comment]
A few changes have been made:
* Updated the flow links so that they use the flow link access validation (they were overwritten fields in the view)
* Moved the summary field to the bottom (like enforcement actions) because it is the least important field
* Unified the saving of advice so that new and updated advice uses the same logic (to avoid code duplication)
* Updated advice edit page title to use the correct verb (add or edit)
* Corrected capitalisation of labels on advise view page
* Updated to use correct issue_date field instead of the date the advice was changed (and updated all legacy advise in hook_update), also corrected formatting of issue date (GDS standard) on the advise view page

File: web/modules/custom/par_data/par_data.install [hunk 5]
```diff
@@ -668,31 +669,31 @@ function par_data_update_8037() {
...
-          '%advice' => $par_data_advice->id(),
-        ];
-        $par_data_advice->getLogger($this->getLoggerChannel())->error($message, $replacements);
-      }
+      $save = TRUE;
     }
     // Set the status.
     if ($par_data_advice->getRawStatus() !== 'deleted') {
       $par_data_advice->setParStatus('active', TRUE);
-      $par_data_advice->set('advice_status', 'active');
```
Comment (jon-edery) [inline web/modules/custom/par_data/par_data.install]: Should not have been removed this is added for a reason.  the default active state for existing advice is now not set in staging . 
Screen Shot 2019-06-30 at 21 11 35

**Pull request merged**